From 4c862fb208fc71618b6fe6147ece899dc55add9d Mon Sep 17 00:00:00 2001
From: Markus Hennecke <markus-hennecke@markus-hennecke.de>
Date: Sat, 2 Sep 2017 17:21:27 +0200
Subject: [PATCH] sunxi: Fix HW address write order

The contents for mac_a1 and mac_a0 registers were swapped in commit ace1520.
---
 drivers/net/sunxi_emac.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/sunxi_emac.c b/drivers/net/sunxi_emac.c
index 37685f30f3..5581c17818 100644
--- a/drivers/net/sunxi_emac.c
+++ b/drivers/net/sunxi_emac.c
@@ -335,8 +335,8 @@ static int _sunxi_write_hwaddr(struct emac_eth_dev *priv, u8 *enetaddr)
 	enetaddr_lo = enetaddr[2] | (enetaddr[1] << 8) | (enetaddr[0] << 16);
 	enetaddr_hi = enetaddr[5] | (enetaddr[4] << 8) | (enetaddr[3] << 16);
 
-	writel(enetaddr_hi, &regs->mac_a1);
-	writel(enetaddr_lo, &regs->mac_a0);
+	writel(enetaddr_lo, &regs->mac_a1);
+	writel(enetaddr_hi, &regs->mac_a0);
 
 	return 0;
 }
